Orange/widgets/data/owdatasets.py

```diff
@@ -477,6 +459,24 @@ def closeEvent(self, event):
         self.header_state = bytes(self.view.header().saveState())
         super().closeEvent(event)
 
+    def ensure_local(self, domain, filename, force=False,
+                     progress_advance=None):
+        localfiles = LocalFiles(self.local_cache_path,
+                                serverfiles=ServerFiles(server=self.INDEX_URL))
+        if force:
+            localfiles.download(domain, filename, callback=progress_advance)
+        return localfiles.localpath_download(
+            domain, filename, callback=progress_advance)
```

Comment:
`ensure_local` as such should not be a method. Parameterize it with target local path and index url if necessary.

Reply:
Changed.